### DISCUSSION TITLE
How to handle a vector (svg/dxf) with compound path?

### DISCUSSION THREAD

**Discussion by RebeccaStevens** [jscad/OpenJSCAD.org#1237] [Q&A]
I've got a bunch of svg icons I want to make into 3d objects via a `extrudeLinear`.
However, most of these icons involve compound paths and I'm not having much luck processing them.

This is how I'm opening the files:

```ts
const fs = require("node:fs");
const svgDeserializer = require("@jscad/svg-deserializer").deserialize;

const rawData = fs.readFileSync("./assets/icon1.svg", { encoding: "utf8" });
const iconData = svgDeserializer({ output: "geometry" }, rawData).map((p) => polygon(p));
```

However, `iconData` ends up being an array of the multiple geometries that make up the compound path; not a single geometry.
I've tried using both `union` and `subtract` to merge the geometries, both before and after then using `extrudeLinear` to make the objects I want, but I can't achieve the desired output.

As part of the debugging process, I performed a `extrudeLinear` on each of the geometries and exported them.
I then imported them into blender to see if I can merge them (with a boolean difference) properly there. However, blender wouldn't merge them properly either unless I recalculated the objects' normals first.

Is there anyway I can recalculate the normals in JSCad to fix this issue?

Also is there a better approach I should be using perform this batch task?

**Comment by andreasplesch** [comment]
I think jscad calculates normals as needed, from polygon points and their winding order. So perhaps the winding orders of the compound paths in the svg are not consistent with each other for being extruded. For a ring, for example, you would need the outer circle wind in the opposite sense of the inner circle. But more specifics would be needed. A screenshot of the undesired output, and the svg of a simple icon. If you export to x3d you can visualize simply here: 

https://create3000.github.io/x_ite/playground/

https://create3000.github.io/x_ite/playground/?url=https://gist.githubusercontent.com/andreasplesch/dc3595c5cf5198bedd7dbdb119a7a059/raw/4c9aed0dd47191873fc4e87870125eecc75c1ab1/jscad.x3d

**Comment by z3dev** [comment]
@RebeccaStevens Welcome. There was a nice long discussion from an issue raised a few months back. Hopefully, you can find a few ideas from this.

https://github.com/jscad/OpenJSCAD.org/issues/1087
